examples/direct_fidelity_estimation.py

```diff
@@ -260,11 +261,13 @@ def direct_fidelity_estimation(circuit: cirq.Circuit, qubits: List[cirq.Qid],
     # above) does bother it, so we re-normalize the probs.
     p /= np.sum(p)
 
-    noisy_simulator = cirq.DensityMatrixSimulator(noise=noise)
     fidelity = 0.0
 
     if samples_per_term == 0:
         # sigma in https://arxiv.org/abs/1104.3835
+        if not isinstance(sampler, cirq.DensityMatrixSimulator):
+            raise TypeError('sampler is not a cirq.DensityMatrixSimulator')
```

Comment:
Test coverage failed. You need a unit test that verifies the exception is raised in the right circumstances.

Reply:
Done.